### DISCUSSION TITLE
Grid Based Movement

### DISCUSSION THREAD

**Discussion by HexSleeves** [excaliburjs/Excalibur#2337] [Q&A]
Curious how one would generate grid based games and movement. For something like a rougelike where each movement is usually defined with either a +1 or -1 to the postiion Vector. Has anyone built something grid based?

**Comment by eonarheim** [comment]
Hi @lecoqjacob 

We recently made a game that did some grid based movement for [Ludum Dare 48](https://github.com/excaliburjs/ludum-48), I extracted the relevant bits and created a new sample! I figure this is going to be a common use that people will want to reference 👍 

https://github.com/excaliburjs/sample-grid

Let me know what you think! The code doing the moving is [here](https://github.com/excaliburjs/sample-grid/blob/main/src/beetle.ts#L70)
